### PR TITLE
Implement resin.models.device.reboot()

### PR DESCRIPTION
**Pull request by jviotti**
This function allows the user to perform a hard-reboot on the device by
using the Supervisor API:

https://github.com/resin-io/resin-supervisor/blob/master/docs/API.md#post-v1reboot

This function, as well as `resin.models.device.restart()` is not tested in
the integration suite since its not a behaviour we can easily assert on.

Fixes: https://github.com/resin-io/resin-sdk/issues/195

Misc changes:

- Clarify that `device.restart()` restarts the application

### CODE REVIEW

File: DOCUMENTATION.md [hunk 1]
```diff
@@ -729,8 +732,12 @@ resin.models.device.move('7cf02a6', 'MyApp', function(error) {
 ```
 <a name="resin.models.device.restart"></a>
 ##### device.restart(uuid) ⇒ <code>Promise</code>
+This function restarts the Docker container running
+the application on the device, but doesn't reboot
+the device itself.
+
 **Kind**: static method of <code>[device](#resin.models.device)</code>  
-**Summary**: Restart device  
+**Summary**: Restart device application
```
Comment (pcarranzav) [inline DOCUMENTATION.md]: Might be better to rephrase this to `Restart application on device`

**Comment by pcarranzav** [comment]
LGTM (besides that one comment)

File: DOCUMENTATION.md (hunk 1 above)
Comment (jviotti) [inline DOCUMENTATION.md]: @pcarranzav Good suggestion, thanks!
